src/main/java/com/thealgorithms/geometry/JarvisMarch.java

```diff
@@ -0,0 +1,147 @@
+            // Check if both references point to the same object
+            if (this == obj) {
+                return true;
+            }
+            // Check if obj is an instance of Point
+            if (!(obj instanceof Point)) {
+                return false;
+            }
+            Point other = (Point) obj;
+            // Compare x and y coordinates for equality
+            return Double.compare(x, other.x) == 0 && Double.compare(y, other.y) == 0;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(x, y); // Generate hash code based on x and y coordinates
+        }
+    }
+
+    /**
+     * Computes the convex hull of a given list of points using the Jarvis March algorithm.
+     *
+     * @param points a list of Points for which to compute the convex hull
+     * @return a list of Points representing the vertices of the convex hull in counter-clockwise order
+     */
+    public static List<Point> jarvisMarch(List<Point> points) {
+        List<Point> hull = new ArrayList<>();
+
+        // If there are less than 3 points, a convex hull cannot be formed
+        if (points.size() < 3) {
+            return hull;
+        }
+
+        // Find the leftmost point (with the smallest x-coordinate)
+        Point leftmost = points.get(0);
+        for (Point p : points) {
+            if (p.getX() < leftmost.getX()) {
+                leftmost = p; // Update leftmost point if a new leftmost point is found
+            }
+        }
+
+        Point current = leftmost; // Start from the leftmost point
+
+        do {
+            hull.add(current); // Add current point to the hull
+
+            Point nextTarget = points.get(0); // Initialize next target as first point in list
+
+            for (Point candidate : points) {
+                // Skip current point
+                if (candidate.equals(current)) {
+                    continue;
+                }
+
+                // Check if candidate makes a left turn or is collinear and farther than nextTarget
+                if (nextTarget.equals(current) || isLeftTurn(current, nextTarget, candidate) || (isCollinear(current, nextTarget, candidate) && distance(current, candidate) > distance(current, nextTarget))) {
+                    nextTarget = candidate; // Update next target if conditions are met
+                }
+            }
+
+            current = nextTarget; // Move to the next target point
+
+        } while (!current.equals(leftmost)); // Continue until we loop back to the starting point
+
+        return hull; // Return the computed convex hull
+    }
+
+    /**
+     * Determines whether moving from point A to point B to point C makes a left turn.
+     *
+     * @param a the starting point
+     * @param b the second point
+     * @param c the third point
+     * @return true if it makes a left turn, false otherwise
+     */
+    private static boolean isLeftTurn(Point a, Point b, Point c) {
```

Comment:
These methods looks like utility methods. So can be moved or to Point class there or utility class.